Orso.Arpa.Domain/UserDomain/Repositories/CustomUserClaimsPrincipalFactory.cs

```diff
@@ -0,0 +1,26 @@
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+using Orso.Arpa.Domain.UserDomain.Model;
+
+namespace Orso.Arpa.Domain.UserDomain.Repositories
+{
+    public class CustomUserClaimsPrincipalFactory : UserClaimsPrincipalFactory<User>
+    {
+        public CustomUserClaimsPrincipalFactory(
+        UserManager<User> userManager,
+        IOptions<IdentityOptions> optionsAccessor)
+        : base(userManager, optionsAccessor)
+        {
+        }
+        protected override async Task<ClaimsIdentity> GenerateClaimsAsync(User user)
+        {
+            ClaimsIdentity claims = await base.GenerateClaimsAsync(user);
+            claims.AddClaim(new Claim("nameid", user.UserName));
+            claims.AddClaim(new Claim("name", user.DisplayName));
+            claims.AddClaim(new Claim("sub", user.Id.ToString()));
+            return claims;
+        }
+    }
```

Comment:
Now the only solution I haven't tried for the claims is adding them into signinmanager
When inspecting SignInManager, I see where IUserClaimsPrincipalFactory is used : https://github.com/dotnet/aspnetcore/blob/7c810658463f35c39c54d5fb8a8dbbfd463bf747/src/Identity/Core/src/SignInManager.cs
For this purpose, I have create a custom sign in manager and change all usages of sign in manager to custom sign in manager